external/esp_idf_port/esp32/esp32_queue_api.c

```diff
@@ -0,0 +1,434 @@
+/******************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include <semaphore.h>
+#include <pthread.h>
+#include <errno.h>
+#include <sched.h>
+#include <debug.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <tinyara/arch.h>
+#include <tinyara/clock.h>
+#include "esp_attr.h"
+#include "esp32_queue_api.h"
+#include "mq_tryreceive.h"
+/************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************/
+#define MAX_QUEUE_INFO 10
```

Comment:
is it necessary to define queue depth as 10? it could be reduced to free some static memory

Reply:
recently, at least 2 queue is created.
for each item, occupy 32B,  all 10 used   320B, not big.